addons/api/addon/models/credential.js

```diff
@@ -23,6 +26,9 @@ export default class CredentialModel extends GeneratedCredentialModel {
    * @type {boolean}
    */
   get isUnknown() {
+    types = this.features.isEnabled('json-credentials')
```

Comment:
Are we intentionally re-setting the `types` variable if `isUnknown` accessed? That seems like it could be prone to unexpected behavior since we're exporting it for use outside the model and it won't change until the `isUnknown` getter is called.

Can't we create the right types on initialization based off the feature flag rather than re-assigning? The consumers of the types downstream then don't need to re-filter or check with the same feature flag

Reply:
The API addon can and probably should know about any available types.  Let's not worry about feature flags within API addon.  This would be separate from any UI related concerns where we hide or show CRUD features based on a feature flag.  So, let's list `json` here with no special handling in the addon.

Reply:
That's fine too, I would also have expected all the types to be available but my main concern was the re-assigning of the `types` variable so either way works for me.

Reply:
This is getting a refactor based on the discussion I had with @randallmorey yesterday. I will ping the team once it is pushed up.